File: src/Python/01_Interakcja_z_konsola/Zad9.py
"""
Otrzymujesz roczna stope procentowa, liczbe lat na splacenie kredytu oraz wysokosc
udzielonego kredytu. Oblicz miesieczna rate oraz calkowity koszt kredytu.
"""

if __name__ == "__main__":

    print(
        "podaj stope procentowa, liczbe lat na splacenie kredytu  oraz wysokosc udzielonego kredytu"
    )

    stopa = float(input())
    lata = int(input())
    kredyt = int(input())

    r = stopa / 12 * 0.01
    n = lata * 12

    rata = kredyt * r * (1 + r) ** n / ((1 + r) ** n - 1)

    c_koszt = rata * n

    print("rata mieieczna to {:0.2f}".format(rata))

    print("calkowity koszt to {:0.2f}".format(c_koszt))

File: src/Cpp/01_Interakcja_z_konsola/Zad9.cpp
#include <cmath>
#include <iomanip>
#include <iostream>

/*
Otrzymujesz roczną stopę procentową, liczbę lat na spłacenie kredytu
oraz wysokość udzielonego kredytu. Oblicz miesięczną ratę oraz
całkowity koszt kredytu.
*/

int main() {

  std::cout
      << "Podaj: stope procentowa, liczbe lat na splacenie kredytu oraz wysokosc 
      udzielonego kredytu " << std::endl;

      float stopa;
  int lata;
  int wysokoscKredytu;

  std::cin >> stopa;
  std::cin >> lata;
  std::cin >> wysokoscKredytu;

  float r = stopa / 12 * 0.01;
  int n = lata * 12;

  float rata = wysokoscKredytu * r * pow((1 + r), n) / (pow((1 + r), n) - 1);
  float calkKoszt = rata * n;

  std::cout << std::fixed << std::setprecision(2);
  std::cout << "rata mieieczna to " << rata << std::endl;
  std::cout << "calkowity koszt to " << calkKoszt << std::endl;

  return 0;
}
